### PR TITLE
[Security] Fix NullToken handling in ExpressionVoter

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 5.3
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Security/Core/Authorization/Voter/ExpressionVoter.php [hunk 1]
```diff
@@ -74,7 +75,7 @@ private function getVariables(TokenInterface $token, $subject): array
 
         $variables = [
             'token' => $token,
-            'user' => $token->getUser(),
+            'user' => $token instanceof NullToken ? '' : $token->getUser(),
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Core/Authorization/Voter/ExpressionVoter.php:78]: getUser returns a UserInterface, but NullToken leads to a string?

**Comment by nicolas-grekas** [comment]
Can you please add a test?
But we need to resolve https://github.com/symfony/symfony/pull/42513/files/077a2385c64ff736d65569aea95c5cc25ae5280c#r691232229 first.

**Comment by chalasr** [comment]
Putting a `Needs Work` status until the discussion around the `NullToken::getUser()` return value in https://github.com/symfony/symfony/pull/42513/files/077a2385c64ff736d65569aea95c5cc25ae5280c#r691232229 is resolved.

**Comment by nicolas-grekas** [comment]
Not needed anymore.

**Pull request closed without merging**